kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/status/NMStatusConverter.java

```diff
@@ -111,6 +151,242 @@ private static void setIP4Status(NetworkInterfaceStatusBuilder<?> builder,
         });
     }
 
+    private static void setWifiStatus(WifiInterfaceStatusBuilder builder, Properties wirelessDeviceProperties,
+            Optional<Properties> activeAccessPoint, List<Properties> accessPoints, String countryCode,
+            List<WifiChannel> supportedChannels) {
+        NM80211Mode mode = NM80211Mode.fromUInt32(wirelessDeviceProperties.Get(NM_DEVICE_WIRELESS_BUS_NAME, "Mode"));
+        builder.withMode(wifiModeConvert(mode));
+
+        List<NMDeviceWifiCapabilities> capabilities = NMDeviceWifiCapabilities
+                .fromUInt32(wirelessDeviceProperties.Get(NM_DEVICE_WIRELESS_BUS_NAME, "WirelessCapabilities"));
+        builder.withCapabilities(wifiCapabilitiesConvert(capabilities));
+        builder.withSupportedRadioModes(wifiRadioModesConvert(capabilities));
+
+        // builder.withSupportedBitrates
+
+        Pair<List<Integer>, List<Long>> kuraSupportedChannels = wifiChannelsConvert(supportedChannels);
+        builder.withSupportedChannels(kuraSupportedChannels.getLeft());
+        builder.withSupportedFrequencies(kuraSupportedChannels.getRight());
+
+        builder.withCountryCode(countryCode);
+
+        if (mode == NM80211Mode.NM_802_11_MODE_AP) {
+            if (!activeAccessPoint.isPresent()) {
+                logger.warn("No access point found for interface in MASTER mode.");
+                return;
+            }
+            // Only one AP should be available in MASTER mode
+            WifiAccessPoint ap = wifiAccessPointConvert(accessPoints.get(0));
+            builder.withActiveWifiAccessPoint(Optional.of(ap));
+        } else {
+            if (activeAccessPoint.isPresent()) {
+                WifiAccessPoint ap = wifiAccessPointConvert(activeAccessPoint.get());
+                builder.withActiveWifiAccessPoint(Optional.of(ap));
+            }
+        }
+
+        builder.withAvailableWifiAccessPoints(wifiAccessPointConvert(accessPoints));
+    }
```

Comment:
Does it make sense to add the available AP even if the interface is configured in MASTER mode?

Reply:
IMO yes: if I want to change the interface from AP to infra mode I already have the available access point to connect to.